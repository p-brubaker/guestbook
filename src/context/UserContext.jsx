import { createContext, useState, useContext, useEffect } from 'react';

const UserContext = createContext();

function UserProvider({ children }) {
  const [user, setUser] = useState(null);

  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
}

function useUser() {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('useUser hook must be called within a UseProvider');
  }

  return context;
}

export { useUser, UserProvider };
