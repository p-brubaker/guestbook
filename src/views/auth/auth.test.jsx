import React from 'react';
import { render, screen } from '@testing-library/react';
import Login from './Auth';
import { UserProvider } from '../../context/UserContext';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

it('should render the Login component', async () => {
  const history = createMemoryHistory();
  history.push('/');

  const { container } = render(
    <UserProvider>
      <Router history={history}>
        <Login />
      </Router>
    </UserProvider>
  );

  screen.getByText('User Name');
  screen.getByText('Password');
  screen.getByRole('button', { name: 'Submit' });
  expect(container).toMatchSnapshot();
});
