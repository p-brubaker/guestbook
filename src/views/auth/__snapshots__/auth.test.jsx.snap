// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Login component 1`] = `
<div>
  <div>
    <form>
      <label
        for="user-input"
      >
        User Name
      </label>
      <input
        id="user-input"
        name="user-input"
        value=""
      />
      <label
        for="password-input"
      >
        Password
      </label>
      <input
        id="password"
        name="password"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
  </div>
</div>
`;
